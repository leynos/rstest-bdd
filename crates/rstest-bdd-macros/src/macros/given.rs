//! Implementation of the `#[given]` macro.

use proc_macro::TokenStream;

/// Macro for defining a Given step that registers with the step inventory.
pub(crate) fn given(attr: TokenStream, item: TokenStream) -> TokenStream {
    super::step_attr(attr, item, crate::StepKeyword::Given)
}
