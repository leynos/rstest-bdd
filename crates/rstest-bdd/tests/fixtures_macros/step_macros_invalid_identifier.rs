//! Compile-fail step definition with an invalid identifier.
use rstest_bdd_macros::given;

#[given("invalid identifier")]
fn 1invalid() {}

fn main() {}
