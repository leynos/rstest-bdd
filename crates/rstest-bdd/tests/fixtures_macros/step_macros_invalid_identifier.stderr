error: expected identifier, found `1invalid`
 --> tests/fixtures_macros/step_macros_invalid_identifier.rs:5:4
  |
5 | fn 1invalid() {}
  |    ^^^^^^^^ expected identifier
  |
help: identifiers cannot start with a number
 --> tests/fixtures_macros/step_macros_invalid_identifier.rs:5:4
  |
5 | fn 1invalid() {}
  |    ^

error: expected identifier
 --> tests/fixtures_macros/step_macros_invalid_identifier.rs:5:4
  |
5 | fn 1invalid() {}
  |    ^^^^^^^^
